libtfhe/core/arithmetic/polynomial_torus_big.cpp

```diff
@@ -0,0 +1,342 @@
+    const TorusPolynomial<BigTorus> *poly2,
+    const PolynomialParams<BigTorus> *params,
+    TfheThreadContext *context,
+    Allocator alloc)
+{
+    // const int32_t N = params->N;
+    // double norm = 0;
+
+    // // Max between the coefficients of abs(poly1-poly2)
+    // for (int32_t i = 0; i < N; ++i) {
+    //     double r = abs(TorusUtils<BigTorus>::to_double(poly1->coefs[i] - poly2->coefs[i], params->zmodule_params));
+    //     if (r > norm) { norm = r; }
+    // }
+    // return norm;
+    return 0;
+}
+
+
+template<>
+void TorusPolynomial<BigTorus>::MultNaive_plain_aux(
+    BigTorus *__restrict result,
+    const INT_TYPE *__restrict poly1,
+    const BigTorus *__restrict poly2,
+    const int32_t N,
+    const ZModuleParams<BigTorus> *zparams,
+    TfheThreadContext *context,
+    Allocator alloc)
+{
+    const int32_t _2Nm1 = 2 * N - 1;
+    BigTorus *ti = alloc.newObject<BigTorus>(zparams, &alloc);
+
+    for (int32_t i = 0; i < N; i++) {
+        BigTorus *ri = result+i;
+        zero(ri, zparams);
+        for (int32_t j = 0; j <= i; j++) {
+            mul(ti, poly1+j, poly2+(i-j), zparams, alloc);
+            add(ri, ri, ti, zparams);
+        }
+    }
+
+    for (int32_t i = N; i < _2Nm1; i++) {
+        BigTorus *ri = result+i;
+        zero(ri, zparams);
+        for (int32_t j = i - N + 1; j < N; j++) {
+            mul(ti, poly1+j, poly2+(i-j), zparams, alloc);
+            add(ri, ri, ti, zparams);
+        }
+    }
+}
+
+template<>
+void TorusPolynomial<BigTorus>::MultNaive_aux(
+    BigTorus *__restrict result,
+    const INT_TYPE *__restrict poly1,
+    const BigTorus *__restrict poly2,
+    const int32_t N,
+    const ZModuleParams<BigTorus> *zparams,
+    TfheThreadContext *context,
+    Allocator alloc)
+{
+    BigTorus *ti = alloc.newObject<BigTorus>(zparams, &alloc);
+
+    for (int32_t i = 0; i < N; i++) {
+        BigTorus *ri = result+i;
+        zero(ri, zparams);
+
+        for (int32_t j = 0; j <= i; j++) {
+            mul(ti, poly1+j, poly2+(i-j), zparams, alloc);
+            add(ri, ri, ti, zparams);
+        }
+        for (int32_t j = i + 1; j < N; j++) {
+            mul(ti, poly1+j, poly2+(N + i - j), zparams, alloc);
+            sub(ri, ri, ti, zparams);
+        }
+    }
+}
+
+/**
+ * This is the naive external multiplication of an integer polynomial
+ * with a torus polynomial. (this function should yield exactly the same
+ * result as the karatsuba or fft version)
+ */
+template<>
+void TorusPolynomial<BigTorus>::MultNaive(
+    TorusPolynomial<BigTorus> *result,
+    const IntPolynomial<INT_TYPE> *poly1,
+    const TorusPolynomial<BigTorus> *poly2,
+    const PolynomialParams<BigTorus> *params,
+    TfheThreadContext *context,
+    Allocator alloc)
+{
+    assert(result != poly2);
+
+    const int32_t N = params->N;
+
+    TorusPolynomial<BigTorus>::MultNaive_aux(result->coefs, poly1->coefs,
+        poly2->coefs, N, params->zmodule_params, context, alloc);
```

Comment:
alloc->child_allocator()

Actually, in this case, the argument will dissapear, but an allocator should not be passable by value.
(-> new issue)